src/main/java/org/mozilla/msrp/platform/vertical/video/YoutubeRepository.kt

```diff
@@ -67,4 +70,20 @@ class YoutubeRepository @Inject constructor(private val youtube: YouTube) {
             emptyList()
         }
     }
+
+    private fun String.compact(): String {
+        val duration = Duration.parse(this)
+
+        val hours: String = duration.toHours().toString()
+        val twoDigitFormatter = DecimalFormat("00")
+        val minutes: String = twoDigitFormatter.format(duration.toMinutes() % TimeUnit.HOURS.toMinutes(1))
```

Comment:
ArithmeticException

Reply:
`ArithmeticException` would throw only if rounding is needed with rounding mode being set to `RoundingMode.UNNECESSARY`. It uses `RoundingMode.HALF_EVEN` by default.